pkg/kubelet/dockertools/manager.go

```diff
@@ -948,6 +955,16 @@ func (dm *DockerManager) KillPod(pod kubecontainer.Pod) error {
 		wg.Add(1)
 		go func(container *kubecontainer.Container) {
 			defer util.HandleCrash()
+
+			// TODO: Handle this without signaling the pod infra container to
+			// adapt to the generic container runtime.
+			if container.Name == PodInfraContainerName {
+				err := dm.networkPlugin.TearDownPod(pod.Namespace, pod.Name, kubeletTypes.DockerID(container.ID))
+				if err != nil {
+					glog.Errorf("Failed tearing down the infra container: %v", err)
+					errs <- err
```

Comment:
Yes. I just noticed that, it seems to be a historical problem. So the buffered size is `len(pod.Containers)`, while we will send at most `len(pod.Containers) + 1` (the 1 came from the tear down) errors. And since no one is receiving the channel, it can block I think?